### PR TITLE
deduplicate shared fields across types

### PR DESCRIPTION
**Pull request by xianny**
# Description

Makes it easier to pass params to shared functions e.g. price comparison utils operates on return types for both meta transactions and swap/quote. 

Easier to review in split view.

# Testing Instructions

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Test changes on the staging environment with [Matcha API Staging](https://api-staging.matcha.xyz).

    -   [ ] SRA/Limit orders
    -   [ ] Swap endpoints
    -   [ ] Meta transaction endpoints
    -   [ ] Depth charts

    For more information see `0x API Matcha smoke test runbook` in Quip.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -532,55 +513,59 @@ export interface GetTransactionRequestParams {
...
-    excludedSources: ERC20BridgeSource[];
-    includedSources?: ERC20BridgeSource[];
-    apiKey: string | undefined;
-    includePriceComparisons: boolean;
-    affiliateFee: PercentageFee;
-    affiliateAddress: string | undefined;
+}
+
+/**
+ * End quote-related tyoes
```
Comment (mzhu25) [inline src/types.ts:563]:
```suggestion
 * End quote-related types
```

**Review by mzhu25 — APPROVED** [review]
lgtm, don't forget to use a conventional commit PR title or whatevs

**Review by kimpers — APPROVED** [review]
Nice! Thanks for cleaning this up 🙏

**Comment by xianny** [comment]
included in https://github.com/0xProject/0x-api/pull/439

**Pull request closed without merging**